### PR TITLE
Update the build script to support opening S.P.Corelib

### PR DESCRIPTION
**Pull request by tannergooding**
This updates the `.\build.cmd -vs ...` command to support `System.Private.Corelib` as a valid option by resolving it against `src\coreclr\src\System.Private.Corelib\System.Private.Corelib.sln`. Previously it would resolve it to `src\libraries\System.Private.Corelib\System.Private.Corelib.sln` which does not exist.

CC. @ViktorHofer

### CODE REVIEW

File: eng/build.ps1 [hunk 1]
```diff
@@ -118,18 +118,30 @@ if ($vs) {
 
   if (-Not (Test-Path $vs)) {
     $solution = $vs
-    # Search for the solution in libraries
-    $vs = Split-Path $PSScriptRoot -Parent | Join-Path -ChildPath "src\libraries" | Join-Path -ChildPath $vs | Join-Path -ChildPath "$vs.sln"
+
+    # Search for the solution in coreclr
+    $vs = Split-Path $PSScriptRoot -Parent | Join-Path -ChildPath "src\coreclr\src" | Join-Path -ChildPath $vs | Join-Path -ChildPath "$vs.sln"
```
Comment (ViktorHofer) [inline eng/build.ps1:123]: What about mono's corelib?

Comment (ViktorHofer) [reply eng/build.ps1:123]: We control that via the RuntimeFlavor switch.

Comment (tannergooding) [reply eng/build.ps1:123]: Where does that solution live? I can add it...

Comment (tannergooding) [reply eng/build.ps1:123]: ah, I found it `src\mono\netcore`

Comment (tannergooding) [reply eng/build.ps1:123]: @ViktorHofer, do we have an existing mechanism to check if any property is `RuntimeFlavor=Mono`, or do I effectively need too do something like `if ($properties.Where({$_ -eq "/p:RuntimeFlavor=Mono"}, 'First').Count -gt 0`?

Comment (ViktorHofer) [reply eng/build.ps1:123]: You would want to add the RuntimeFlavor as a powershell param at the root of the build.ps1 script and then condition on it. We haven't yet needed the differentiation so you are the first one who stumbles upon this.

Comment (tannergooding) [reply eng/build.ps1]: Should be done. I also added it to `build.sh` for parity.

**Review by ViktorHofer — APPROVED** [review]
Assuming you tested this locally, LGTM. thanks

**Pull request merged**